src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -43,28 +46,30 @@ public static DynamicAnalysisInjector TryCreate(MethodSymbol method, BoundStatem
                 return null;
             }
 
-            MethodSymbol createPayload = GetCreatePayload(methodBodyFactory.Compilation, methodBody.Syntax, diagnostics);
+            MethodSymbol createPayloadForMethodsSpanningSingleFile = GetCreatePayloadForMethodsSpanningSingleFile(methodBodyFactory.Compilation, methodBody.Syntax, diagnostics);
+            MethodSymbol createPayloadForMethodsSpanningMultipleFiles = GetCreatePayloadForMethodsSpanningMultipleFiles(methodBodyFactory.Compilation, methodBody.Syntax, diagnostics);
 
             // Do not instrument any methods if CreatePayload is not present.
-            if ((object)createPayload == null)
+            if ((object)createPayloadForMethodsSpanningSingleFile == null || (object)createPayloadForMethodsSpanningMultipleFiles == null)
             {
                 return null;
             }
 
             // Do not instrument CreatePayload if it is part of the current compilation (which occurs only during testing).
             // CreatePayload will fail at run time with an infinite recursion if it is instrumented.
-            if (method.Equals(createPayload))
+            if (method.Equals(createPayloadForMethodsSpanningSingleFile) || method.Equals(createPayloadForMethodsSpanningMultipleFiles))
             {
                 return null;
             }
 
-            return new DynamicAnalysisInjector(method, methodBody, methodBodyFactory, createPayload, diagnostics, debugDocumentProvider, previous);
+            return new DynamicAnalysisInjector(method, methodBody, methodBodyFactory, createPayloadForMethodsSpanningSingleFile, createPayloadForMethodsSpanningMultipleFiles, diagnostics, debugDocumentProvider, previous);
         }
 
-        private DynamicAnalysisInjector(MethodSymbol method, BoundStatement methodBody, SyntheticBoundNodeFactory methodBodyFactory, MethodSymbol createPayload, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, Instrumenter previous)
+        private DynamicAnalysisInjector(MethodSymbol method, BoundStatement methodBody, SyntheticBoundNodeFactory methodBodyFactory, MethodSymbol createPayloadForMethodsSpanningSingleFile, MethodSymbol createPayloadForMethodsSpanningMultipleFiles, DiagnosticBag diagnostics, DebugDocumentProvider debugDocumentProvider, Instrumenter previous)
```

Comment:
Sure I will split the lines that you have pointed out...  However, this is a general problem in this file (i.e. there are many long lines that were present before my changes including lines that I haven't touched in this PR 😄).